source/Cosmos.System2_Plugs/System/IO/FileImpl.cs

```diff
@@ -119,19 +119,14 @@ public static string[] ReadAllLines(string aFile)
 
             return result;
         }
-
-        public static void WriteAllLines(string aFile, IEnumerable<string> contents)
+        
+        public static void WriteAllLines(string aFile, string[] contents)
         {
-            string text = null;
-
-            foreach (var line in contents)
-            {
-                text = string.Concat(text, line, Environment.NewLine);
-            }
-
+            string text = String.Join(Environment.NewLine, contents);
+            
             Global.mFileSystemDebugger.SendInternal("Writing contents");
             Global.mFileSystemDebugger.SendInternal(text);
-
+            
```

Comment:
Nit: trim whitespace.

Reply:
Working when trying to use the codes in the kernel :)

Reply:
Means it works with Writing Test and Reading Test

Reply:
You need to add a test in the FatTestRunner so AppVoyer will compile it in any PR and we known automatically that it did not broke.

TestRunner is easier for you too as now you need to have an external kernel, compile Cosmos anytime, Compile the Kernel... right?

Reply:
Yes, You mean I Need to Create A Tester in this pull request